pages/posts/github-case-study.md

```diff
@@ -0,0 +1,80 @@
+
+More specifically, some page content is assembled fully automatically from JSON data files and some JSON is manually authored by content writers.
+Within the platform, the application's entire queryable in-memory context is retrievable as JSON.
+
+"Not being able to validate our JSON data against a JSON schema would result in bugs in production like missing data on automated documentation pages or unavailable pages", explain Rachael Sewell and Robert Sese, software engineers who work on GitHub's Docs Engineering team.
+
+Without proper schemas for their JSON data, there was no way to verify whether code changes introduced new bugs, or to ensure that data consumed from external sources was in the format needed for automated documentation pages.
+
+<figure className="mt-10">
+  <img src="/blog/img/posts/2023/github-case-study/working.webp"/>
+  <figcaption className="mt-2 mb-10 text-sm text-center text-gray-500">Hubbers (GitHub staff) working to empower developers.</figcaption>
+</figure>
+
+## Solution
+
+The team began to introduce JSON Schemas to validate all of the JSON data files, context data and API request bodies consumed or produced as part of the application.
+Each time a change to the data model is made, the schemas are updated accordingly, and in any circumstance the schemas are used to validate that the application functions correctly with each change.
+
+This occurs in three main ways:
+
+  * While the application runs in production, each API call has its request body validated via a corresponding JSON Schema before passing the event off to a data warehouse
+  * When retrieving external data in automation pipelines, whenever data is transformed from a source format into JSON, the generated data is validated against a JSON Schema to ensure it has been correctly transformed.
+    If validation succeeds, the generated data is checked into a git repository to be used in production.
+    Otherwise, the failure is raised for investigation.
+  * When running continuous integration each time a change is made to the application, various additional schemas ensure:
+
+    * that YAML frontmatter properties are correctly included in Markdown files which are used to generate pages within the application
+    * that YAML or JSON data files which contain page content and which have been hand authored by content writers are correctly formed
+    * that the context object created at runtime which contains the entirety of the site content along with a site tree for the application is itself correctly formed
+
+<p className="text-2xl my-10">"Choosing JSON schema to allow JSON schema validation was a natural and obvious choice that our team made. It has been a fundamental part of our application since we moved from having a static site to a dynamic application about 3 years ago." - Rachael Sewell & Robert Sese, Docs Engineers at GitHub</p>
```

Comment:
Does the `<p className="text-2xl my-10">` match the styling of block quotes in other posts?  It'd be good to maintain consistency.

Reply:
It seems like yes, see e.g. https://github.com/json-schema-org/blog/blob/ae73ee9f108e8cf3a468898c4cbaf31e120351d3/pages/posts/tyler-technologies-case-study.md?plain=1#L52